resources/views/pages/admin/link-teacher-student.blade.php

```diff
@@ -0,0 +1,182 @@
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+
+  <!-- Modal -->
+  <div class="modal fade" id="myModal" role="dialog">
+    <div class="modal-dialog">
+    
+      <!-- Modal content-->
+      <div class="modal-content">
+        <div class="modal-header">
+          <button type="button" class="close" data-dismiss="modal">&times;</button>
+          <h4 class="modal-title">{{ __('linking.student-koppelen') }}</h4>
+        </div>
+        <div class="modal-body">
+
+        <div class="panel panel-default">
+            <div class="panel-heading">
+                <h3 class="panel-title">{{ __('linking.stap-1') }}</h3>
+            </div>
+            <div class="panel-body">
+                <p id="chosenTeacher"></p>
+           </div>
+        </div>  
+        <div class="panel panel-default">
+            <div class="panel-heading">
+                <h3 class="panel-title">{{ __('linking.stap-2') }}</h3>
+            </div>
+            <div class="panel-body">
+            <div class="dropdown">
+                    <div id="myDropdown" class="dropdown-content">
+                        <input type="text" placeholder="{{ __('linking.placeholder') }}" id="myInput" onkeyup="filterFunction()">
+                        <div class="dropdown-links">
+                        @foreach($students as $student)
+                        @if($student->userlevel === 0)
+                        <a onclick="chooseStudent({{ $student->student_id }})">{{ $student->studentnr }} - {{ $student->firstname }} {{ $student->lastname }}</a>
+                        @endif
+                        @endforeach
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>
+
+        <div class="panel panel-default">
+            <div class="panel-heading">
+                <h3 class="panel-title">{{ __('linking.stap-3') }}</h3>
+            </div>
+            <div class="panel-body">
+                <select id="selectWPLP" class="form-control" name="wplp">
+                    <option>{{ __('linking.kies-stage') }}</option> 
+                </select>
+            </div>
+        </div>
+
+        
+         {!! Form::open(array(
+            'url' =>  route('update-workplacelearningperiod')))
+            !!}
+            <div class="form-group">
+                <input id='wplp' type='text', name='wplp_id' class="form-control">
+            </div>
+                          
+            <div class="form-group">
+                <input type='text', name='teacher_id' id="teacher_id" class="form-control">
+            </div>
+            
+
+        <div class="modal-footer">
+            {{ Form::submit('Koppelen', array('class' => 'btn btn-primary')) }}
+            {{ Form::close() }}
+        </div>
+      </div>
+      
+    </div>
+  </div>
+  
+</div>
+
+<?php
+
+$allStudents = array();
```

Comment:
* This logic should happen in the controller, not in the view.
* The student class has user level methods `isStudent` `isTeacher` `isAdmin`. Use those instead of checking the actual `userlevel` value

I assume you are creating a new array for these because they are `Collection` instances? If that is the case, you can use the `all()` [method on a collection](https://laravel.com/docs/5.7/collections#method-all) to convert it to a pure array. On the same `Collection` instances you can also use the `filter` method to filter on the role.

Reply:
I tried it, but I don't get the logic in the controller with the use of isStudent. Do you have an example of this?

Reply:
This should help you I think

```php
public function __invoke(Request $request)
    {
        // ->all()->all() looks confusing, but Laravel called "Collection to Array" method all() for some reason
        $wplperiods = $this->wplpRepository->all()->all();
        $workplaces = $this->workplaceRepository->getAll()->all();

        // The all method is not yet present in this repository, so you'll need to add it. It can be something like
        // return Student::sortable()->get(); 
        // The sortable() call allows you to still use the sortable table columns in the view
        /** @var Collection|Student[] $users */
        $users = $this->studentRepository->all();

        $students = $users->filter(static function (Student $user) {
            return $user->isStudent();
        })->all();

        $teachers = $users->filter(static function (Student $user) {
            return $user->isTeacher();
        })->all();
        
        return view('pages.admin.link-teacher-student', [
            'students'   => $students,
            'teachers'   => $teachers,
            'wplperiods' => $wplperiods,
            'workplaces' => $workplaces,
        ]);
    }
```